parser/src/lexer/compat.rs

```diff
@@ -0,0 +1,910 @@
+                    (python_token, enderpy_token, "Missing token".to_string())
+                }
+                TokenMismatch::WrongKind(python_token, enderpy_token) => {
+                    let message = format!(
+                        "Wrong token kind.\nExpected: {:?}\nActual: {:?}",
+                        python_token.kind, enderpy_token.kind
+                    );
+                    (Some(python_token), Some(enderpy_token), message)
+                }
+                TokenMismatch::WrongValue(
+                    python_token,
+                    enderpy_token,
+                    expected_value,
+                    actual_value,
+                ) => (
+                    Some(python_token),
+                    Some(enderpy_token),
+                    format!(
+                        "Wrong token value.\nExpected: {:?}\nActual: {:?}",
+                        expected_value, actual_value
+                    ),
+                ),
+                TokenMismatch::WrongStartEnd(
+                    python_token,
+                    enderpy_token,
+                    expected_start,
+                    expected_end,
+                    actual_start,
+                    actual_end,
+                ) => (
+                    Some(python_token),
+                    Some(enderpy_token),
+                    format!(
+                        "Wrong token start/end offset.\nExpected: {:?} - {:?}\nActual: {:?} - {:?}",
+                        expected_start, expected_end, actual_start, actual_end,
+                    ),
+                ),
+            };
+            if include_all_tokens {
+                row.extend_from_slice(&[
+                    python_tokens
+                        .iter()
+                        .map(|token| {
+                            let is_this_token =
+                                python_token.as_ref().is_some_and(|tok| tok == token);
+                            format!("{}{:?}", if is_this_token { "→ " } else { "" }, token)
+                        })
+                        .collect::<Vec<String>>()
+                        .join("\n"),
+                    enderpy_tokens
+                        .iter()
+                        .map(|token| {
+                            let is_this_token =
+                                enderpy_token.as_ref().is_some_and(|tok| tok == token);
+                            format!("{}{:?}", if is_this_token { "→ " } else { "" }, token)
+                        })
+                        .collect::<Vec<String>>()
+                        .join("\n"),
+                    message,
+                ]);
+            } else {
+                row.extend_from_slice(&[
+                    python_token.map_or("None".to_string(), |t| format!("{:?}", t)),
+                    enderpy_token.map_or("None".to_string(), |t| format!("{:?}", t)),
+                    message,
+                ]);
+            }
+            if include_source {
```

Comment:
I'd say to print the source at the top or bottom if the there are any mismatches(without the env var setting or on by default), it helps a lot to figure out the where the issue happens.
Right now it's shown in a column, that's fine but I think seeing the source only once top or bottom should be good.
![image](https://github.com/Glyphack/enderpy/assets/20788334/9688f9cb-e12d-4b8b-8a25-1288315546fa)

Reply:
Added source to the top, before the table is printed.